lib/storage/client.js

```diff
@@ -1182,6 +1183,267 @@ class Client extends CryptoConsumer {
+      }
+    )
+    // Check the response code
+    switch (response.status) {
+      case 204:
+        return true
+      case 403:
+        throw new Error('Unauthorized')
+      default:
+        throw new Error('Error while deleting record data.')
+    }
+  }
+  /**
+   * listGroupMembers lists members of the TozStore Group given.
+   *
+   * @param {string} groupId  UUID assigned by TozStore, used to identify a group.
+   *
+   * @returns {GroupMember[]} A response from TozStore; the group members of the group.
+   */
+  async listGroupMembers(groupId) {
+    // Call the API endpoint to remove group members
+    let response = await this.authenticator.tsv1Fetch(
+      `${this.config.apiUrl}/v2/storage/groups/${groupId}/members`,
+      {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      }
+    )
+    let groupMembersJson = await validateResponseAsJSON(response)
+    let listGroupMembers = []
+    groupMembersJson.forEach(member => {
+      listGroupMembers.push(member)
+    })
+    return listGroupMembers
+  }
+  /**
+   * listRecordsSharedWithGroup
+   *
+   * @param {string} groupId   UUID assigned by TozStore, used to identify a group.
+   * @param {Array} writerIds  Array of writerIds to filter responses by; optional.
+   * @param {number} nextToken  Indicates where to start pagination; optional.
+   * @param {number} max        The maximum number of groups to list per request; optional.
+   *
+   * @return {Array}  An array of records returned from TozStore.
+   */
+  async listRecordsSharedWithGroup(
+    groupId,
+    writerIds = [],
+    nextToken = null,
+    max = null
+  ) {
+    const listSharedRequest = {
+      group_id: groupId,
+      writer_ids: writerIds,
+      nextToken: nextToken,
+      max: max,
+    }
+    let encodedUrl = urlEncodeData(listSharedRequest)
+    let response = await this.authenticator.tsv1Fetch(
+      `${this.config.apiUrl}/v2/storage/groups/${groupId}/share?${encodedUrl}`,
+      {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      }
+    )
+    let listSharedWithGroups = []
+    let recordsJson = await validateResponseAsJSON(response)
+    if (recordsJson.results == null) {
+      return listSharedWithGroups
+    }
+    for (var i = 0; i < recordsJson.results.length; i++) {
+      let meta = await new Meta(
+        recordsJson.results[i].meta.writer_id,
+        recordsJson.results[i].meta.user_id,
+        recordsJson.results[i].meta.type,
+        recordsJson.results[i].meta.plain
+      )
+      let recordData = await new RecordData(recordsJson.results[i].record_data)
+      let record = await new Record(
+        meta,
+        recordData,
+        recordsJson.results[i].rec_sig
+      )
+      let decryptedRecord = shared.decryptRecord(this, record)
+      listSharedWithGroups.push(decryptedRecord)
+    }
+    return listSharedWithGroups
```

Comment:
This methods needs to return the next token, so the caller of this method can paginate through his results